src/backend/distributed/commands/table.c

```diff
@@ -222,11 +236,33 @@ PostprocessCreateTableStmtForeignKeys(CreateStmt *createStatement)
 		List *relationFKeyCreationCommands =
 			GetForeignConstraintCommandsInternal(relationId, nonDistTableFKeysFlag);
 		DropRelationForeignKeys(relationId, nonDistTableFKeysFlag);
-		ExecuteAndLogDDLCommandList(relationFKeyCreationCommands);
+
+		bool skip_validation = true;
+		ExecuteAndLogForeignKeyCreateCommandList(relationFKeyCreationCommands,
+												 skip_validation);
 	}
 }
 
 
+/*
+ * ShouldEnableLocalReferenceForeignKeys is a wrapper around getting the GUC
+ * EnableLocalReferenceForeignKeys. If the coordinator is not added
+ * to the metadata, the function returns false. Else, the function returns
+ * the value set by the user
+ *
+ */
+static bool
+ShouldEnableLocalReferenceForeignKeys(void)
+{
+	if (!CoordinatorAddedAsWorkerNode())
+	{
+		return false;
+	}
+
+	return EnableLocalReferenceForeignKeys;
```

Comment:
maybe check this first since it's a cheaper check
```c
if (!EnableLocalReferenceForeignKeys)
{
    return false;
}

return CoordinatorAddedAsWorkerNode();
```